synapse/lib/cell.py

```diff
@@ -2118,6 +2118,13 @@ async def handoff(self, turl, timeout=30):
         Hand off leadership to a mirror in a transactional fashion.
         '''
         _dispname = f' ahaname={self.conf.get("aha:name")}' if self.conf.get('aha:name') else ''
+
+        if not self.isactive:
+            mesg = f'HANDOFF: {_dispname} is not the current leader and cannot receive the promotion handoff' \
+                   f' from {s_urlhelp.sanitizeUrl(turl)}.'
+            logger.error(mesg)
+            raise s_exc.BadState(mesg=mesg, turl=turl, cursvc=_dispname)
```

Comment:
Do we want to use a new & specific exception here?

Reply:
BadState seems like a reasonable exception to raise in this situation

Reply:
Nah, ``BadState`` seems correct :+1: